packages/core/contracts/l2/polygon/OracleChildTunnel.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./tunnel/FxBaseChildTunnel.sol";
+import "../../oracle/interfaces/OracleAncillaryInterface.sol";
+import "../../oracle/interfaces/RegistryInterface.sol";
+import "./OracleBaseTunnel.sol";
+
+/**
+ * @title Adapter deployed on sidechain to give financial contracts the ability to trigger cross-chain price requests to 
+ * the mainnet DVM. Also has the ability to receive published prices from mainnet. This contract can be treated as the 
+ * "DVM" for this network, because a calling contract can request and access a resolved price request from this 
+ * contract.
+ * @dev The intended client of this contract is an OptimisticOracle on sidechain that needs price
+ * resolution secured by the DVM on mainnet.
+ */
+contract OracleChildTunnel is OracleBaseTunnel, OracleAncillaryInterface, FxBaseChildTunnel {
+    constructor(address _fxChild, address _finderAddress)
+        OracleBaseTunnel(_finderAddress)
+        FxBaseChildTunnel(_fxChild)
+    {}
+
+    // This assumes that the local network has a Registry that resembles the mainnet registry.
+    modifier onlyRegisteredContract() {
+        RegistryInterface registry = RegistryInterface(finder.getImplementationAddress(OracleInterfaces.Registry));
+        require(registry.isContractRegistered(msg.sender), "Caller must be registered");
+        _;
+    }
+
+    /**
+     * @notice This should be called to bridge a price request to mainnet.
+     * @dev Can be called only by a registered contract that is allowed to make DVM price requests. Will mark this
+     * price request as Requested, and therefore able to receive the price resolution data from mainnet. Emits a message
+     * that will be included in regular checkpoint of all sidechain transactions to mainnet.
+     * @param identifier Identifier of price request.
+     * @param time Timestamp of price request.
+     * @param ancillaryData extra data of price request.
+     */
+    function requestPrice(
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData
+    ) public override onlyRegisteredContract() {
+        // This implementation allows duplicate price requests to emit duplicate MessageSent events via 
```

Comment:
Important to read this design decision. This is based on feedback from Polygon team and is less of a concern than `publishPrice` because of the `onlyRegisteredContract` modifier.